src/Parser/MarksmanshipHunter/Modules/Spells/Trueshot.js

```diff
@@ -0,0 +1,40 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+
+class QuickShot extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  trueShotCDReduction = 0;
+  totalFromCD = 0;
+
+  get tsAvailableCasts() {
+    const player = this.combatants.selected;
+    const quickShotRank = player.traitsBySpellId[SPELLS.QUICK_SHOT_TRAIT.id];
+
+    if (quickShotRank === 0) {
+      return;
```

Comment:
Returning here would return I think `undefined` which would mess up with the calculation in CastEfficiency. I understand that there's most likely a 0 probability of someone without this trait would be using the analyzer but it still should be accounted for (my suggestion below should solve this).